### DISCUSSION TITLE
Logging Throttling Events and Backoff Times

### DISCUSSION THREAD

**Discussion by bajce** [pnp/pnpcore#1315] [Q&A] [answered]
We are currently using the PnP Core SDK in our solution and have encountered a situation where we suspect our requests are being throttled by Microsoft. Although we understand the SDK handles throttling internally, for our monitoring and debugging purposes, we require an explicit log of when throttling occurs and the advised wait/backoff time.

Could you please advise if there's a way to configure the SDK to log:
1. Instances when the requests are throttled.
2. The wait/backoff time suggested by Microsoft if available.

This information would be helpful for our operation's transparency and efficiency. If this feature is not currently supported, could you consider it as a feature request?

Thank you for your assistance in advance!

**Comment by jansenbe** [comment, marked as answer]
If you enable "Information" logging this should show up: 
https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Services/Core/Http/RateLimiter.cs#L128
https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Services/Core/Http/RetryHandlerBase.cs#L111

**Reply by bajce** [reply]
Should 'Information' logging be only enabled for PnP package or in general? I want to avoid unnecessary logging. 

Here is as example of how I plan to configure it in my solution:
```csharp
.ConfigureLogging(logging =>
 {
     logging.ClearProviders();
     logging.AddFilter("PnP", LogLevel.Information);
 })
.UseNLog()
```
and in my NLog.config:
`<logger name="PnP.*" minlevel="Info" writeTo="pnp-file" />`

From the links you provided im not sure if the logging statements will show up with PnP filter.
